src/js/requestAwareMiddleware.js

```diff
@@ -0,0 +1,45 @@
+/*
+
+    Middleware that creates a `gpii.express.handler` component and uses that to process each request.  See the
+    documentation for details:
+
+    https://github.com/GPII/gpii-express/blob/master/docs/middleware.md
+
+ */
+"use strict";
+var fluid = require("infusion");
+
+fluid.defaults("gpii.express.middleware.requestAware", {
+    gradeNames: ["gpii.express.middleware"],
+    timeout: 5000, // The default timeout we will pass to whatever grade we instantiate.
+    distributeOptions: [
+        {
+            source: "{that}.options.timeout",
+            target: "{that gpii.express.handler}.options.timeout"
+        },
+        {
+            source: "{that}.options.handlerGrades",
+            target: "{that gpii.express.handler}.options.gradeNames"
+        }
+    ],
+    method:     "get",
+    events: {
+        "onRequest": null
+    },
+    dynamicComponents: {
+        requestHandler: {
```

Comment:
See comments on contentTypeAware for reforming this to be a single standard definition